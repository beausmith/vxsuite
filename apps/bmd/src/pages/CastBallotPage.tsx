import React from 'react'
import styled from 'styled-components'

import LinkButton from '../components/LinkButton'
import Main, { MainChild } from '../components/Main'
import Prose from '../components/Prose'
import Text from '../components/Text'

const Directions = styled.div`
  margin: auto;
  & h1 {
    text-align: center;
  }
`

const Instructions = styled.ol`
  margin: 0 auto;
  border-radius: 0.25rem;
  background: #ffffff;
  max-width: 24rem;
  padding: 1.5rem 1rem;
  list-style: none;
  @media (min-width: 480px) {
    padding: 2.5rem;
  }
  & > li {
    position: relative;
    margin-bottom: 1.75rem;
    padding-left: 1.75rem;
    counter-increment: asdf;
  }
  & > li:last-child {
    margin-bottom: 0;
  }
  & > li::before {
    position: absolute;
    top: 0;
    left: 0;
    font-size: 1.25rem;
    font-weight: 700;
    content: counter(asdf) '.';
  }
`

const StartPage = () => {
  return (
    <Main>
      <MainChild center>
        <Directions>
          <Prose>
            <h1 aria-label="Verify and Cast Your Ballot.">
              Verify and Cast Your Ballot
            </h1>
            <Text center>Retrieve your printed ballot from the printer.</Text>
            <Instructions>
              {/* <li>
                <h2>Collect Printed Ballot.</h2>
                <p>The printer has printed your ballot.</p>
              </li> */}
              <li>
                <h2>Verify Ballot Selections.</h2>
                <p>Review and confirm all selections on your printed ballot.</p>
              </li>
              <li>
                <h2>Cast in Ballot Box.</h2>
                <p>Deposit your ballot into the secured ballot box.</p>
              </li>
            </Instructions>
            <Text center>
              I have verified my selections and will cast my ballot.
            </Text>
            <Text center>
              <LinkButton primary to="/">
                Start Over
              </LinkButton>
            </Text>
          </Prose>
        </Directions>
      </MainChild>
    </Main>
  )
}

export default StartPage
