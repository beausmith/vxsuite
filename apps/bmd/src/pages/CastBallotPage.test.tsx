import { axe } from 'jest-axe'
import React from 'react'
import { Route } from 'react-router-dom'

import { render } from '../../test/testUtils'

import CastBallotPage from './CastBallotPage'

it(`renders CastBallotPage`, async () => {
  const { container } = render(<Route path="/" component={CastBallotPage} />, {
    route: '/',
  })
  expect(container.firstChild).toMatchSnapshot()
  expect(await axe(container.innerHTML)).toHaveNoViolations()
})
