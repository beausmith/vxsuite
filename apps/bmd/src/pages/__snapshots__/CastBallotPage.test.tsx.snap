// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CastBallotPage 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
  padding: 1rem 0.5rem 2rem;
}

.c1 {
  width: 100%;
  max-width: 35rem;
  margin: auto auto;
}

.c3 {
  line-height: 1.2;
  max-width: 66ch;
}

.c3 h1 {
  line-height: 1.1;
  margin: 2rem 0 1rem;
}

.c3 h2 {
  margin: 1.5rem 0 1rem;
}

.c3 h3,
.c3 p {
  margin: 1rem 0;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p,
.c3 h3 + p {
  margin-top: -0.75rem;
}

.c3 :first-child {
  margin-top: 0;
}

.c3 :last-child {
  margin-bottom: 0;
}

.c6 {
  box-sizing: border-box;
  cursor: pointer;
  background: #4caf50;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: white;
  line-height: 1;
  white-space: nowrap;
}

.c4 {
  text-align: center;
}

.c2 {
  margin: auto;
}

.c2 h1 {
  text-align: center;
}

.c5 {
  max-width: 24rem;
  list-style: none;
  margin: 0 auto;
  padding: 2.5rem;
  background: white;
  border-radius: 0.25rem;
}

.c5 > li {
  margin-bottom: 1.75rem;
  position: relative;
  counter-increment: asdf;
  padding-left: 1.75rem;
}

.c5 > li:last-child {
  margin-bottom: 0;
}

.c5 > li:before {
  content: counter(asdf) '.';
  position: absolute;
  top: 0;
  left: 0;
  font-weight: bold;
  font-size: 1.5rem;
}

@media (min-width:480px) {
  .c0 {
    padding: 2rem 1rem;
  }
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media print {
  .c1 {
    max-width: 100%;
    margin: 0;
  }
}

@media (min-width:480px) {
  .c3 {
    line-height: 1.3;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1>
          Verify and Cast Ballot
          <span
            class="visually-hidden"
          >
            .
          </span>
        </h1>
        <p
          class="c4"
        >
          Retrieve your printed ballot from the printer.
        </p>
        <ol
          class="c5"
        >
          <li>
            <h2>
              Verify Ballot Selections.
            </h2>
            <p>
              Review and confirm all selections on your printed ballot.
            </p>
          </li>
          <li>
            <h2>
              Cast in Ballot Box.
            </h2>
            <p>
              Fold your ballot and deposit into the secured ballot box.
            </p>
          </li>
        </ol>
        <p
          class="c4"
        >
          I have verified and will cast my ballot.
        </p>
        <p
          class="c4"
        >
          <button
            class="c6"
            type="button"
          >
            Start Over
          </button>
        </p>
      </div>
    </div>
  </div>
</main>
`;
